components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2146,6 +2153,8 @@ private IdentityProvider getIDP(Connection dbConnection, String idPName, int idp
         } catch (ConnectorException e) {
             throw new IdentityProviderManagementException("Error occurred while retrieving the identity connector " +
                     "configurations.", e);
+        } catch (SecretManagementException e) {
+            throw new IdentityProviderManagementException("Error while retrieving secrets of IDP.", e);
```

Comment:
Improve the error message by adding more context. ex: idp name, which tenant domain

Reply:
check other instances and update them as well

Reply:
Resolved with [2b47e12](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/2b47e121b1c14f3ddce774f6b6a867c957fb5848)